### PR TITLE
SS Conversion: Atoms, Machines, n' Mobs

### PR DESCRIPTION
**Pull request by tigercat2000**
This converts the machine and mob processes to the SMC. Additionally, it
adds the Atom subsystem, which handles all Initialize() calls in place
of the old gameticker. Due to incompatibility with our atmospherics
(FUCK OUR ATMOSPHERICS FOR FUCKING EVER JESUS CHRIST WHO THE FUCK MADE
THIS PIECE OF GODDAMN SHIT) atmospherics machines do not use
Initialize() as they should, instead opting for a custom atmos_init
proc that the air controller handles.

### CODE REVIEW

File: code/ATMOSPHERICS/atmospherics.dm [hunk 1]
```diff
@@ -29,28 +28,21 @@ Pipelines + Other Objects -> Pipe network
 	var/pipe_color
 	var/obj/item/pipe/stored
 	var/image/pipe_image
-	var/global/datum/pipe_icon_manager/icon_manager
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/atmospherics.dm]: I have this great idea. Let's make a global datum, attached to all machines and when it initializes, it'll make a NEW icon manager datum, then if it already exists, we'll just return. This totally makes sense! Also, if you happen to delete it, it'll just kill the server with the lag it generates not to mention generate hundreds of runtimes!

Comment (KasparoVy) [reply code/ATMOSPHERICS/atmospherics.dm]: Absolute stroke of genius

File: code/game/atoms.dm [hunk 2]
```diff
@@ -42,6 +42,62 @@
 
 	var/admin_spawned = 0	//was this spawned by an admin? used for stat tracking stuff.
 
+	var/initialized = FALSE
+
+/atom/New(loc, ...)
+	if(use_preloader && (src.type == _preloader.target_path))//in case the instanciated atom is creating other atoms in New()
+		_preloader.load(src)
+	. = ..()
+	attempt_init(...)
```
Comment (Fox-McCloud) [inline code/game/atoms.dm]: what about 
```
	var/list/created = SSatoms.created_atoms
	if(created)
		created += src
```?

File: code/game/machinery/camera/presets.dm [hunk 3]
```diff
@@ -81,7 +81,7 @@
 	assembly.upgrades.Add(new /obj/item/assembly/prox_sensor(assembly))
 	setPowerUsage()
 	// Add it to machines that process
-	machine_processing |= src
+	START_PROCESSING(SSmachines, src)
```
Comment (Fox-McCloud) [inline code/game/machinery/camera/presets.dm]: 👌

File: code/game/machinery/cloning.dm [hunk 4]
```diff
@@ -230,7 +230,8 @@
 	if(!R.dna)
 		R.dna = new /datum/dna()
 
-	var/mob/living/carbon/human/H = new /mob/living/carbon/human(src, R.dna.species)
+	var/mob/living/carbon/human/H = new /mob/living/carbon/human(src)
```
Comment (Fox-McCloud) [inline code/game/machinery/cloning.dm]: make sure to look for this elsehwere; pretty sure others utilize this ability.

File: code/modules/error_handler/error_handler.dm [hunk 5]
```diff
@@ -96,7 +96,7 @@ var/total_runtimes_skipped = 0
 
 	// Now to actually output the error info...
 	log_world("\[[time_stamp()]] Runtime in [e.file],[e.line]: [e]")
-	log_runtime_txt("\[[time_stamp()]] Runtime in [e.file],[e.line]: [e]")
+	log_runtime_txt("Runtime in [e.file],[e.line]: [e]")
```
Comment (Fox-McCloud) [inline code/modules/error_handler/error_handler.dm]: Uhh?

File: code/modules/power/power.dm [hunk 6]
```diff
@@ -232,26 +232,6 @@
 					. += C
 	return .
 
-/hook/startup/proc/buildPowernets()
```
Comment (Fox-McCloud) [inline code/modules/power/power.dm]: hooks..... *gags*

File: code/modules/research/server.dm [hunk 7]
```diff
@@ -23,7 +23,7 @@
 	component_parts += new /obj/item/stack/cable_coil(null,1)
 	component_parts += new /obj/item/stack/cable_coil(null,1)
 	RefreshParts()
-	initialize(); //Agouri
+	initialize_serv(); //Agouri // fuck you agouri
```
Comment (Fox-McCloud) [inline code/modules/research/server.dm]: Hue.

File: code/game/machinery/camera/presets.dm (hunk 3 above)
Comment (KasparoVy) [reply code/game/machinery/camera/presets.dm]: 👌 ✖️ 💯

File: code/game/atoms.dm [hunk 8]
```diff
@@ -42,6 +42,62 @@
 
 	var/admin_spawned = 0	//was this spawned by an admin? used for stat tracking stuff.
 
+	var/initialized = FALSE
+
+/atom/New(loc, ...)
+	if(use_preloader && (src.type == _preloader.target_path))//in case the instanciated atom is creating other atoms in New()
```
[the remark this answers is not in the document]
Comment (tigercat2000) [reply code/game/atoms.dm]: I don't really wanna debug the maploader and convert this entire var over to GLOB...

File: code/game/atoms.dm (hunk 2 above)
Comment (tigercat2000) [reply code/game/atoms.dm]: I have no idea what that tracker is for, I didn't port most of SSatoms. I figured it'd be a bad idea to add a new global list of all atoms without some use for it.

File: code/game/machinery/cloning.dm (hunk 4 above)
Comment (tigercat2000) [reply code/game/machinery/cloning.dm]: I looked over every human/New call I could find but I didn't see anything else -- should I readd my debugging message to human/New to throw warnings if anything calls it with more than a loc argument?

File: code/modules/error_handler/error_handler.dm (hunk 5 above)
Comment (tigercat2000) [reply code/modules/error_handler/error_handler.dm]: rust-g automatically adds timestamps now, doing this results in duplicated timestamps.

File: code/game/machinery/cloning.dm (hunk 4 above)
Comment (Fox-McCloud) [reply code/game/machinery/cloning.dm]: Yeah, I would, *just* in case.

File: code/modules/error_handler/error_handler.dm (hunk 5 above)
Comment (Fox-McCloud) [reply code/modules/error_handler/error_handler.dm]: right-o.

File: code/game/atoms.dm (hunk 2 above)
Comment (Fox-McCloud) [reply code/game/atoms.dm]: Right, we'll wait until we get the entire thing ported over then.

File: code/game/atoms.dm (hunk 8 above)
[the remark this answers is not in the document]
Comment (Fox-McCloud) [reply code/game/atoms.dm]: Reee

File: code/game/machinery/cloning.dm (hunk 4 above)
Comment (tigercat2000) [reply code/game/machinery/cloning.dm]: Done

Comment (Fox-McCloud) [reply code/game/machinery/cloning.dm]: I knew there was another one---there's one under monkey cubes.

**Comment by SkullyRoberts** [comment]
By the way, @tigercat2000 This PR entirely broke wires. Just figured you'd want to know.

**Comment by KasparoVy** [comment]
We don't need no wires where we're going comrades

**Pull request merged**